test/reporters/list.spec.js

```diff
@@ -138,10 +138,16 @@ describe('List reporter', function() {
       var actual = {a: 'actual'};
       var expected = {a: 'expected'};
       var checked = false;
+      var stringifyDiffs = 'stringifyDiffObjs';
       var err;
       test = {};
-      runner.on = runner.once = function(event, callback) {
-        if (!checked && event === 'fail') {
+      runner = createMockRunner('fail', 'fail', null, null, test);
+      runner.on = function(event, callback) {
+        if (
+          callback.toString().includes(stringifyDiffs) && // target correct fail event callback
```

Comment:
Might also migrate this to the end of the `if` condition. It's much more expensive than the other two conditionals.

Reply:
Yep will do